src/Npgsql/NpgsqlConnector.Auth.cs

```diff
@@ -366,11 +366,17 @@ class AuthenticationCompleteException : Exception { }
             if (password != null)
                 return password;
 
-            // No password was provided. Attempt to pull the password from the pgpass file.
-            var passFilePath = Settings.Passfile ?? PostgresEnvironment.PassFile ?? PostgresEnvironment.PassFileDefault;
-            if (passFilePath != null && File.Exists(passFilePath))
+            var passFile = Settings.Passfile ?? PostgresEnvironment.PassFile;
+            if (passFile is null &&
+                PostgresEnvironment.PassFileDefault is { } passFileDefault &&
```

Comment:
nit: we don't usually use this strange-looking syntax to check not-null (see `passFile != null` below here), note that C# 9 [will apparently introduce](https://devblogs.microsoft.com/dotnet/welcome-to-c-9-0/#logical-patterns) `is not null`. Maybe we should just do `!= null` until then - but am OK either way.

Reply:
This expression takes a value from `PassFileDefault`, passes it into `passFileDefault` and checks it for not being `null`, but yeah, we could change it later.

Reply:
Yeah, it's fine - just talking about the syntax `x is { } y` as opposed to `x != y` or `x is not null` (in C# 9). I'm good with whatever here.